DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/RefreshUIDProcessor.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class RefreshUIDProcessor : IAnonymizerProcessor
+    {
+        public Dictionary<string, string> ReplacedUIDs { get; } = new Dictionary<string, string>();
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            DicomUID uid;
+            var old = ((DicomElement)item).Get<string>();
+
+            if (ReplacedUIDs.ContainsKey(old))
+            {
+                uid = new DicomUID(ReplacedUIDs[old], "Anonymized UID", DicomUidType.Unknown);
```

Comment:
So when replacing UID, we are anonymizing the description to "Anonymized  UID" and type to Unknown?